### ISSUE TITLE
Find an alternative to Vercel hobby plan used for PR/main doc publishing

### ISSUE THREAD

**Issue by 3flex** [detekt/detekt#7584]
## Expected Behavior
Multiple detekt maintainers can manage doc publishing.

## Current Behavior
Any changes to Vercel configuration need to be approved by Nico.

## Context
detekt has multiple maintainers but only Nico can change Vercel configuration due to use of the hobby plan.

See #7550 for more.

**Comment by cortinico** [comment]
Agree that this is not ideal:

> Any changes to Vercel configuration need to be approved by Nico.

To be fair. The only settings we need to change on Vercel's dashboard is the Node version. We change that less than once a year.

All the other changes applied via the `vercel.json` file can be deployed by any maintainer (provided that it's on an detekt/detekt fork).

Anyway, the solution here could be to create an account on the `@detekt.dev` email and move the plan over there.

**Comment by 3flex** [comment]
> All the other changes applied via the vercel.json file can be deployed by any maintainer (provided that it's on an detekt/detekt fork).

So it didn't build in my PR because it wasn't a branch in detekt/detekt?

I think Vercel forces node updates periodically anyway so that's probably not actually a task that needs to be done in the config.

This isn't a high priority for anyone and like I said I think when GitHub eventually releases its own solution we move to that, I just wanted to open an issue so we don't forget.

**Comment by cortinico** [comment]
> So it didn't build in my PR because it wasn't a branch in detekt/detekt?

Yeah correct. I had to approve it because it was coming from your personal fork.

> I think Vercel forces node updates periodically anyway so that's probably not actually a task that needs to be done in the config.
> 
> This isn't a high priority for anyone and like I said I think when GitHub eventually releases its own solution we move to that, I just wanted to open an issue so we don't forget.

Agree 👍

**Comment by cortinico** [comment]
I've explored CloudFlare briefly here [#8956](https://github.com/detekt/detekt/pull/8956).

The approach in that PR won't work because the Cloudflare token won't be available for fork PRs.

The alternative would be to use the CloudFlare app. But that also won't work. The problem is that similarly to Vercel, we would have to build the docs inside CloudFlare infra. There we would need to install a JDK, and CloudFlare doesn't support it.

One alternative could be to rewrite the whole `detekt-generator` in Javascript/Node so that ca be executed wherever. That has some benefits as we'll have a better time moving between vendors, but also is quite of a work.

Ideally GH would have its own 1st party solution, based on GH Actions and that would solve all our problems. Till that day we should either:

1. Stay with Vercel, and have previews only for branches (as I'm the only one allowed to approve fork previews).
2. Move all the `detekt-generator` to JS and move to a different vendor.

I'm open to alternatives if someone has ideas.

**Comment by BraisGabin** [comment]
Detekt generator needs to parse kotlin files. I'm not sure if that's easily doable with js.

Also, if we move to js, could we instead use kotlin/js?

**Comment by cortinico** [comment]
> Detekt generator needs to parse kotlin files. I'm not sure if that's easily doable with js.
> Also, if we move to js, could we instead use kotlin/js?

So using Kotlin/JS is 'doable' but it wouldn't really solve the problem.
Say we update `detekt-generator` to target Kotlin/JS, it would still need a Gradle invocation to generate the `.js` file that we then run, to generate the `.md` files used by docusaurus.

We could commit the `detekt-generator/build/dist/js/.../*.js` files, but that's quite of an ugly solution... we would still need contributors to update that build files to see the preview URLs.

Also we'll probably retain most of the `detekt-generator` as is (as today is also used to generate default configs, etc.), we'll just have to adapt the markdown generation capability.
